IngressMaxFields.user.js

```diff
@@ -133,47 +132,152 @@ function wrapper() {
-	data += "<div id='form_area'>";
-	data += "    <textarea class='form_area' name='portal_list_area' rows='30' cols='70' placeholder='Copy and paste portal list here OR upload portal list 	file below. Proper formatting guidelines can be found in the instructions. Anything after a # is considered a comment and will be ignored - be sure to remove any # or ; that appear in a portal name. Each portal should start on a new line.'>" + o.join("\n") + "</textarea>";
-	data += "</div>";
-	data += "<div id='form_part2'>";
-	data += "<div id='file_upload'>";
-	data += "  <br/>";
-	data += "  <label class='upload_button' hidden><span><input id='hidden' type='file' name='portal_list'></span></label><input type='hidden' id='path' placeholder='No file selected' disabled>";
-	data += "</div>";
-	data += "<div id='num_agents'>";
-	data += "    <table width='100%'>";
-	data += "    <tr><td width='50%'>Number of agents:</td><td width='50%'><input type='number' class='num_agents' name='num_agents' value='1' min='1' required></td></tr>";
-	data += "    <tr><td>Use Google Maps?</td><td><input type='checkbox' name='useGoogle' value='YES' checked></td></tr>";
-	data += "    <tr><td>Color scheme</td><td><input type='radio' name='color' value='ENL' checked>ENL</input><input type='radio' name='color' value='RES'>RES</input></td></tr>";
-	data += "    </table>";
-	data += "</div>";
-	data += "<div id='submit'>";
-	data += "    <table>";
-	data += "						<tr><td>Email:</td><td><input type='email' name='email' placeholder='(optional)'></td></tr>";
-	data += "    <tr><td></td><td><input type='submit' class='submit' name='submit' value='Submit!'></td></tr>";
-	data += "    </table>";
-	data += "</div>";
-	data += "</div>";
-	data += "</form>";
+
+    // Return markup for the bookmarks to show in the IMF dialog
+    self.renderPortalBookmarkFolders = function renderPortalBookmarkFolder(folders) {
+        var data = "<p>With Portal Bookmarks, you can populate the list with a folder you have created instead:</p>";
+
+        for (var folder in folders) {
+            data += `<div onClick='window.plugin.ingressmaxfield.appendBookmarkFolder("${folder}");'>${folders[folder]}</div>`
+        }
+
+        return data;
+    }
+
+    // Generate string for given bookmarked portal
+    self.genStrFromBookmarkPortal = function genStrFromBookmarkPortal(portal, folder) {
+        var portalsList = JSON.parse(localStorage['plugin-bookmarks']);
+        var portal = portalsList.portals[folder]["bkmrk"][portal];
+        var latlng = portal.latlng.split(',');
+
+        return self.genStr(portal.label, latlng[0], latlng[1], portal.guid);
+    }
+
+    // Generate string for all the portals
+    self.genStrFromBookmarkFolder = function genStrFromBookmarkFolder(folder) {
+        var data = "",
+            portalsList = JSON.parse(localStorage['plugin-bookmarks']);
+
+        for (var portal in portalsList.portals[folder]["bkmrk"]) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/stefan-nygren/IITC-Ingress-Maxfields-Exporter/file/2651002189/issues/source?bid=3039347&fileBranchId=3324533#l161)